src/antares/tsgen/ts_generator.py

```diff
@@ -52,10 +48,56 @@ class ThermalCluster:
     po_law: ProbabilityLaw
     po_volatility: float
 
+    def __post_init__(self) -> None:
+        _check_outage_gen_params(self)
+
+
+@dataclass
+class ThermalCluster:
+    # available units of the cluster
+    # unit_count: int
+    outage_gen_params: OutageGenerationParameters
+    # nominal power
+    nominal_power: float
+    # modulation of the nominal power for a certain hour in the day (between 0 and 1)
+    modulation: IntArray
+
+    # forced and planed outage parameters
```

Comment:
I think you can remove this part of commented code as you moved these data inside the `OutageGenerationParameters` class